static/app/views/issueDetails/groupDetails.tsx

```diff
@@ -273,6 +270,19 @@ function useEventApiQuery(
   return isLatest ? latestEventQuery : otherEventQuery;
 }
 
+function useSyncGroupStore() {
+  const queryClient = useQueryClient();
+  const router = useRouter();
+  const allGroups = useLegacyStore(GroupStore);
+
+  const params = router.params;
+  const group = allGroups.find(({id}) => id === params.groupId) as Group;
+
+  useEffect(() => {
+    setApiQueryData(queryClient, [`/issues/${params.groupId}/`], group);
```

Comment:
You'll want the query key here to _exactly_ match the query key that you used to make the initial request. So you'll need to include the environments query string as well. I'd recommend creating a function like `makeFetchGroupQueryKey` that you can use for both places.